libc/pthread_cond_test_functions.c

```diff
@@ -70,6 +72,9 @@ void *decrement_count_timed_wait_fail_too_short_timeout(void *args)
 	thread_err_t *thread_err = (thread_err_t *)args;
 	struct timespec time;
 	clock_gettime(CLOCK_REALTIME, &time);
+	/* to set past time and avoid overflow */
+	time.tv_sec = 0;
+	time.tv_nsec = 0;
```

Comment:
I think now both functions `decrement_count_timed_wait_fail_incorrect_timeout` and `decrement_count_timed_wait_fail_too_short_timeout` do the same thing, maybe remove one test? Also I think there's no point in calling `clock_gettime` if the next line sets the time to 0.